provision-runners/setup-runner.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+[[ ! -z "$RUNNER_ADMIN_TOKEN" ]] && echo "RUNNER_ADMIN_TOKEN Not empty" || echo "RUNNER_ADMIN_TOKEN Empty"
+[[ ! -z "$AMI_ID" ]] && echo "AMI_ID Not empty" || echo "AMI_ID Empty"
+[[ ! -z "$AWS_ACCESS_KEY_ID" ]] && echo "AWS_ACCESS_KEY_ID Not empty" || echo "AWS_ACCESS_KEY_ID Empty"
+[[ ! -z "$AWS_SECRET_ACCESS_KEY" ]] && echo "AWS_SECRET_ACCESS_KEY Not empty" || echo "AWS_SECRET_ACCESS_KEY Empty"
+[[ ! -z "$AWS_DEFAULT_REGION" ]] && echo "AWS_DEFAULT_REGION Not empty" || echo "AWS_DEFAULT_REGION Empty"
+
+set -e
+
+echo "setting up dependencies"
+sudo apt -qq update
+sudo apt -qq install jq curl unzip moreutils -y &> /dev/null
+
+#Only install awscli if it's missing (running act)
+if [ ! -x "$(command -v aws)" ]
+  then 
+    echo "aws-cli was not found - installing now"
+    curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+    unzip -qq awscliv2.zip
+    ./aws/install
+  else echo "aws-cli was found on system"
+fi 
+
+#acquire runner token
+echo "Creating runner token for ${GITHUB_REPOSITORY}"
+curl -s -X POST \
```

Comment:
If this command fails (e.g., wrong credentials), the script will continue because curl exits with exit code 0.
If you add flag `--fail-with-body` it will fail with error code 22 for most errors.

```
--fail-with-body

(HTTP) Return an error on server errors where the HTTP response code is 400 or greater). In normal cases when an HTTP server fails to deliver a document, it returns an HTML document stating so (which often also describes why and more). This flag will still allow curl to output and save that content but also to return error 22.

This is an alternative option to --fail which makes curl fail for the same circumstances but without saving the content. 
```

Reply:
I've added that option, there is also an error check on line 32 that will prevent moving forward without a token.